### DISCUSSION TITLE
DevOps

### DISCUSSION THREAD

**Discussion by BryanDollery** [danielmiessler/Fabric#619] [Show and tell]
[I've done something interesting (I think).](https://github.com/BryanDollery/devops-fabric-ai) I'm just playing around with it right now, but I'm a senior software engineer/enterprise-architect/systems-guy/programmer and I've been playing a lot on my new Raspberry Pi 5. It has an arm processor and comes with a-arch64 which I'm quite happy with (running sway for the win).

So, five seconds after becoming completely obsessed with Fabric, I started trying to use it to help me configure the Pi, and save me from endless web searches and reading through blogs. This is the outcome of tinkering for a week or so. There is an example output from my 'script' command in my repo generated from a previous iteration of the prompt.

I don't know if I'm over-engineering this, or if this approach is valuable to anyone else. Personally, it has saved me hours already.

**Comment by superjunaid** [comment]
Bryan, that was a really neat way to use fabric! I’ve gotta dig in and see how it’ll work on my machine and maybe I can build out a version that serves my purpose as I look to integrate this for my podcast post production process.

Where the variable is always the transcript, but the output has a very similar structure to what I want to do with the transcript.

Wondering if you’d be open to chat?

Junaid

**Reply by BryanDollery** [reply]
Happy to chat. This is all interesting.

My approach is to have an alias that calls a function that wraps the call to fabric and which uses one of my own patterns. The function may be the bit you're referring to. You could easily write one that takes a url, passes it through yt to get the transcript, caches it in your 2nd-brain so that you don't have to transcribe something twice, then uses that output with your prompt and the context pattern to generate your next response. Then you could chain more transformations together on the command line, or through custom scripting.

**Reply by BryanDollery** [reply]
Another approach you could try is perhaps doing your entire process in a single pattern/context, like I have with the 'script' pattern. Set up personas, processes, etc. and describe your entire workflow. Then get the ai to perform it. I found that in order to make it work properly, you have to tell it explicitly to output the steps in the process and the conversations between team members. I used a manager to distribute the work and had to carefully break down the incremental nature of the process. Without the output, the AI bypasses the process and jumps straight to an answer, which is fine for really short simple scripts, but doesn't work when things get more complicated and require more steps. It seems that being forced to log the process actually forces the process to be executed.

**Comment by superjunaid** [comment]
Thanks Bryan!The single process does make it a lot more simpler in a complex way but we get to reap the benefits of the output.I’ll start breaking down my process to see what I need and build out the SOPSent from my iPhoneOn Jun 18, 2024, at 11:37 AM, Bryan ***@***.***> wrote:﻿
Another approach you could try is perhaps doing your entire process in a single prompt, like I have with the script. Set up personas, processes, etc. and describe your entire workflow. Then get the ai to perform it. I found that in order to make it work properly, you have to tell it explicitly to output the steps in the process and the conversations between team members. I used a manager to distribute the work and had to carefully break down the incremental nature of the process. Without the output, the AI bypasses the process and jumps straight to an answer, which is fine for really short simple scripts, but doesn't work when things get more complicated and require more steps. It seems that being forced to log the process actually forces the process to be executed.

**Reply by BryanDollery** [reply]
Hey, experimenting last night, I the output of my 'script' pattern to be way too verbose. To the point where gpt is just stopping before the "project" is half finished, and outputting up until that point. It regularly just stops half-way through, say, a code-block, not even closing the fence at the end.

Right now, it's just producing simple scripts, using a single iterative/incremental process. I want to keep that pattern, but I think I'm going to have to do it via multiple calls to the ai, which means adding a lot of complexity to manage the process, and splitting out each of the 'personas' into separate sub-patterns (I can probably use a composition pattern to pre-process the pattern that fabric sees). Rather than having the ai do everything in one pass, I'll have my 'manager' get the stories from the analyst, then I'll loop (in bash) over the stories, passing them to the engineering team (ai) one at a time, storing the code until finally I'll add an extra pass for merging/refactoring the separate branches I've collected into a single coherent code-base.

I'll give it a go tonight and see how it turns out.

**Comment by gchingon** [comment]
did you make it private or take it down? the repo is no longer available
